modules/claims_api/spec/lib/claims_api/error/error_handler_spec.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'claims_api/special_issue_mappers/bgs'
+require 'token_validation/v2/client'
+require 'claims_api/error/error_handler'
+# require 'spec_helper'
+
+describe ApplicationController, type: :controller do # rubocop:disable RSpec/MultipleDescribes
+  let(:scopes) { %w[claim.write] }
+
+  controller do
+    skip_before_action :authenticate
+
+    def test_invalid_token
+      raise ::Common::Exceptions::TokenValidationError
+    end
+  end
+
+  before do
+    routes.draw do
+      get 'test_invalid_token' => 'anonymous#test_invalid_token'
+    end
+  end
+
+  it 'catches an invalid token' do
+    with_okta_user(scopes) do |auth_header|
+      # Following the normal headers: auth_header pattern fails due to
+      # this rspec bug: https://github.com/rspec/rspec-rails/issues/1655
+      # This is the recommended workaround from that issue thread.
+      request.headers.merge!(auth_header)
+
+      get :test_invalid_token
+
+      expect(response).to have_http_status(:unauthorized)
+      parsed_body = JSON.parse(response.body)
+      expect(parsed_body['errors'].size).to eq(1)
+      expect(parsed_body['errors'][0]['title']).to eq('Token Validation Error')
+      expect(parsed_body['errors'][0]['detail']).to eq('Token Validation Error')
+    end
+  end
+end
+
+describe ApplicationController, type: :request do
+  let(:veteran_id) { '1013062086V794840' }
+  let(:scopes) { %w[claim.write] }
+  let(:get_poa_path) { "/services/benefits/v2/veterans/#{veteran_id}/power-of-attorney" }
+
+  # it 'catches an invalid token' do
+  #   with_okta_user(scopes) do |auth_header|
+  #     allow(JWT).to receive(:decode).and_raise(JWT::DecodeError)
+
+  #     get get_poa_path, headers: auth_header
+
+  #     parsed_body = JSON.parse(response.body)
+  #     expect(parsed_body['errors'][0]['detail']).to eq('Invalid token.')
+  #   end
+  # end
+
+  it 'catches an expired token' do
+    with_okta_user(scopes) do |auth_header|
+      allow_any_instance_of(OpenidApplicationController).to receive(:jwt?).and_return(true)
+      allow_any_instance_of(Token).to receive(:valid_issuer?).and_return(true)
+      allow_any_instance_of(Token).to receive(:public_key).and_return(nil)
+      allow(JWT).to receive(:decode).and_raise(JWT::ExpiredSignature)
```

Comment:
@jw81 this simplifies the mocking logic while also getting the expired signature error to get rescued properly.

I think the reason that using just this last line doesn't work is because [`ExpiredSignature` is a child class of `DecodeError`](https://github.com/jwt/ruby-jwt/blob/a79bb0490f66985bcd3a72ba88d502f6615bf3c3/lib/jwt/error.rb#L9), so it gets rescued early in [`jwt?`](https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/controllers/openid_application_controller.rb#L325-L330)